src/lib/db.ts

```diff
@@ -724,4 +754,97 @@ export class DB {
     }
     return this.documentListeners[documentKey];
   }
+
+  private monitorWorkspaceVisibility = (workspace: WorkspaceModelType) => {
+    if (this.workspaceModelDisposers[workspace.sectionId]) {
+      this.workspaceModelDisposers[workspace.sectionId]();
+    }
+    const { user } = this.stores;
+    const updateRef = this.ref(this.getSectionDocumentPath(user, workspace.sectionId));
+    const disposer = (onSnapshot(workspace, (newWorkspace) => {
+      updateRef.update({
+        visibility: newWorkspace.visibility
+      });
+    }));
+    this.workspaceModelDisposers[workspace.sectionId] = disposer;
+  }
+
+  private updateGroupUserSectionDocumentListeners(sectionId: string) {
+    const { user, groups } = this.stores;
+    const userGroup = groups.groupForUser(user.id);
+    const groupUsers = userGroup && userGroup.users;
+    if (groupUsers) {
+      groupUsers.forEach((groupUser) => {
+        if (groupUser.id === user.id) {
+          return;
+        }
+        const currentSectionDocsListener = this.getOrCreateGroupUserSectionDocumentListeners(sectionId, groupUser.id)
+          .sectionDocsRef;
+        if (currentSectionDocsListener) {
+          currentSectionDocsListener.off();
+        }
+        const groupUserSectionDocsRef = this.ref(this.getSectionDocumentPath(user, sectionId, groupUser.id));
+        this.getOrCreateGroupUserSectionDocumentListeners(sectionId, groupUser.id)
+          .sectionDocsRef = groupUserSectionDocsRef;
+        groupUserSectionDocsRef.on("value", (snapshot) => {
```

Comment:
I could get away without using any `once`s here since it's not important for group info to be correct before resolving.